cmd/node/main.go

```diff
@@ -818,27 +821,21 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 	ef.SetTpsBenchmark(tpsBenchmark)
 	ef.SetConfig(efConfig)
 
-	wg := sync.WaitGroup{}
-	go ef.StartBackgroundServices(&wg)
-	wg.Wait()
+	log.Trace("starting background services")
+	ef.StartBackgroundServices()
```

Comment:
The problem was with the starting of backgroundservices on a goroutine?

Reply:
Not quite, the ef.StartBackgroundServices that started on a go routing had and wg.Add on the first line. Starting the rest API web server is blocking so wg.Done will never be called. Depending on how fast the go routine was started, the first instruction after that would either have been wg.Add or wg.Wait. If wg.Wait would have triggered first, since there was no prior wg.Add call, it would have returned immediately, otherwise it would wait indefinitely.